### DISCUSSION TITLE
JURASSIC WORLD - printing multiple cards after creating a new card

### DISCUSSION THREAD

**Discussion by lindseysatterfield** [nss-evening-cohort-14/evening-client-side#195] [Help Me!] [answered]
Lindsey Satterfield, Gabby Tobermann, Tad Sekeres, John Maple

- Yes, we have searched other tickets. Saw #166 and tried the solutions but they did not work

- Whenever we sign in/out and we do not refresh the page, when we create a card, it creates multiples of that card. In Firebase, the duplicates appear with empty strings in the key/value pairs but with new Firebase keys. 

- No error messages are appearing in VS Code or the console

We have googled:
https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/removeEventListener
https://firebase.google.com/docs/auth/web/google-signin
https://developer.mozilla.org/en-US/docs/Web/API/Location/reload
Tried googling for our problem but there were a lot of pages that did not quite touch our issue. Most of the things that we found were not for JavaScript and were more for jQuery issues.

What have you tried?
- we tried `document.querySelector('body').removeEventListener('click', domEvents);` in our `signMeOut()` function
- we tried to add `document.querySelector('body').removeEventListener('click', navigationEvents);`
- we tried adding both ^^^ at the same time
- we tried `window.location.reload();` also in the `signMeOut()` function
- we tried `e.preventDefault()` in the `navigationEvents()`

Links:
https://github.com/nss-evening-cohort-14/jurrasic-world-dashboard-jurrasic-world-dashboard/blob/gt-rides-delete/src/javascripts/components/logoutButton.js
https://github.com/nss-evening-cohort-14/jurrasic-world-dashboard-jurrasic-world-dashboard/blob/gt-rides-delete/src/javascripts/components/loginButton.js
https://github.com/nss-evening-cohort-14/jurrasic-world-dashboard-jurrasic-world-dashboard/blob/gt-rides-delete/src/javascripts/components/logoutButton.js
https://github.com/nss-evening-cohort-14/jurrasic-world-dashboard-jurrasic-world-dashboard/blob/gt-rides-delete/src/javascripts/helpers/events/navEvents.js

**Comment by Gabrielle-Tobermann** [comment, marked as answer]
So update: We had to rewrite our domEvents stuff so that in the addEventlistener we passed a function name instead of an anonymous function. We did that so that the removeEventListener synthax would match exactly the addEventListener. So now in our domEvents module we have two functions: 
- domEventListeners is a function which holds all of our if statements. 
- domEvents just contains our addEventListener like this: 
`domEvents = () => document.querySelector('body').addEventListener('click', domeEventListener);`
Then, in our logout button we have: 
`domEvents = () => document.querySelector('body').removeEventListener('click', domeEventListener);`

**Reply by TrinityChristiana** [reply]
@Gabrielle-Tobermann @lindseysatterfield Is this solution working?

**Reply by Gabrielle-Tobermann** [reply]
It is working on John's branch but we are about to fetch it and check one more time before we merge.

**Reply by Gabrielle-Tobermann** [reply]
It's working!
